adoc/chapters/programming_interface.adoc

```diff
@@ -1422,1390 +1422,1969 @@ info::context::atomic_fence_scope_capabilities
-      Use [code]#info::kernel_device_specific::work_group_size# to query this
-      limit.
+'''
 
-a@
-[source]
+.[apidef]#info::device::max_work_item_sizes#
+[source,role=synopsis,id=api:info-device-max-work-item-sizes]
 ----
-info::device::max_num_sub_groups
+namespace sycl::info::device {
+template<int Dimensions = 3>
+struct max_work_item_sizes<Dimensions> {
+  using return_type = range<Dimensions>;
+};
+} // namespace sycl::info::device
 ----
 
-    @ [.code]#uint32_t#
-   a@ Returns the maximum number of sub-groups in a work-group for any kernel executed on the device.  The minimum value is 1.
+_Remarks:_ Template parameter to [api]#device::get_info#.
 
-a@
-[source]
-----
-info::device::sub_group_sizes
-----
+_Constraints_: Available only when [code]#Dimensions# is 1, 2, or 3.
 
-    @ [.code]#std::vector<size_t>#
-   a@ Returns a [code]#std::vector# of [code]#size_t# containing the set of sub-group sizes supported by the device.
+_Returns:_ The maximum number of work-items that are permitted in a work-group
+for a kernel running in an index space of [code]#Dimensions# dimensions.
+When the device type is not [api]#info::device_type::custom#, the minimum value
+returned from this query is: (1) when [code]#Dimensions# is 1, (1, 1) when
+[code]#Dimensions# is 2, and (1, 1, 1) when [code]#Dimensions# is 3.
 
-a@
-[source]
+'''
+
+.[apidef]#info::device::max_work_group_size#
+[source,role=synopsis,id=api:info-device-max-work-group-size]
 ----
-info::device::preferred_vector_width_char
-info::device::preferred_vector_width_short
-info::device::preferred_vector_width_int
-info::device::preferred_vector_width_long
-info::device::preferred_vector_width_float
-info::device::preferred_vector_width_double
-info::device::preferred_vector_width_half
+namespace sycl::info::device {
+struct max_work_group_size {
+  using return_type = size_t;
+};
+} // namespace sycl::info::device
 ----
 
-    @ [.code]#uint32_t#
-   a@ Returns the preferred native vector width size for built-in scalar types that can be put into vectors. The vector width is defined as the number of scalar elements that can be stored in the vector. Must return 0 for [code]#info::device::preferred_vector_width_double# if the [code]#device# does not have [code]#aspect::fp64# and must return 0 for [code]#info::device::preferred_vector_width_half# if the [code]#device# does not have [code]#aspect::fp16#.
+_Remarks:_ Template parameter to [api]#device::get_info#.
 
-a@
-[source]
-----
-info::device::native_vector_width_char
-info::device::native_vector_width_short
-info::device::native_vector_width_int
-info::device::native_vector_width_long
-info::device::native_vector_width_float
-info::device::native_vector_width_double
-info::device::native_vector_width_half
-----
+_Returns:_ The maximum number of work-items that this device is capable of
+executing in a work-group.
+The minimum value is 1.
+This value is an upper limit and will not necessarily maximize performance.
```

Comment:
Should it be a `note`?

Reply:
This text comes from OpenCL (see [CL_DEVICE_MAX_WORK_GROUP_SIZE](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_DEVICE_MAX_WORK_GROUP_SIZE)), where is is also normative.  I think it makes sense for it to be normative here too.

Reply:
>This value is an upper limit and will not necessarily maximize performance.

It seems to be a weird norm to me. But ok